### PR TITLE
Take into account IV values on initial transfer

### PR DESCRIPTION
**Pull request by andythorne**
### Short Description:

Read `pokemon_potential` config and take into account when running an initial transfer
### Changes:
- Read `config.pokemon_potential` in initial transfer worker and compare with each pokemon.

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Nice catch, surprised this was overlooked originally.

:+1: 

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/115/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by RedSparr0w** [comment]
also maybe for the pokemon under the IV & CP limit you could work out CP \* IV to figure out the best option to keep.

for examples sake:

if you have 3 x Bulbasaur
1. 90 CP & 80% IV
2. 110 CP & 60% IV
3. 120 CP & 60% IV

calculate CP \* IV to decide which one to keep
1. 90 \* 0.8 = 72
2. 100 \* 0.6 = 60
3. 120 \* 0.6 = 72

Keep No.1 as works out better overall (No.3 is just as good based on calculations but No.1 has higher IV which should be priority?)

**Comment by MaxLeiter** [comment]
I think that CP isn't very important running the bot, as you generate so much stardust (and capture plenty of pokemon for the candies)

**Comment by andythorne** [comment]
I agree with @MaxLeiter, I'm much more concerned about IV than CP.

**Comment by MaxLeiter** [comment]
@andythorne think you forgot what branch you're on :P

**Comment by andythorne** [comment]
ah feck, let me just fix that...

**Comment by andythorne** [comment]
phew, there we go!

File: pokemongo_bot/cell_workers/initial_transfer_worker.py [hunk 1]
```diff
@@ -14,10 +14,18 @@ def work(self):
...
+            else:
+                logger.log('[x] Will NOT transfer anything above CP {} or these {}'.format(
+                    self.config.cp, ignlist))
         else:
-            logger.log('[x] Preparing to transfer all Pokemon duplicates, keeping the highest CP of each one type.')
+            if self.config.pokemon_potential:
+                logger.log('[x] Will NOT transfer anything above IV {} or these {}'.format(
+                    self.config.pokemon_potential, ignlist))
+            else:
+                logger.log('[x] Preparing to transfer all Pokemon duplicates, keeping the highest CP of each one type.')
```
Comment (jtdroste) [inline pokemongo_bot/cell_workers/initial_transfer_worker.py]: Could you change lines 16 -> 28 to:

```
        if not self.config.pokemon_potential and not self.config.cp:
            logger.log('[x] Preparing to transfer all Pokemon duplicates, keeping the highest CP of each one type.')
        else:
            logger.log('[x] Will NOT transfer anything above CP {}, IV {} or these {}'.format(
                self.config.cp, self.config.pokemon_potential, ignlist)) 
```

That way we don't have 3 branches of this code doing the same thing :)

**Comment by jtdroste** [comment]
@MaxLeiter I blame drinking + approving commits.  They do not go together :P 

Anyway, added a comment on something I think you should change.  There might be a better way of handling that, I just used the logic that was already there

**Comment by andythorne** [comment]
Good spot @jtdroste. Updated :)

I've refactored to use @wchill's new event system as well.

**Comment by wchill** [comment]
👍 looks good to me.

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/115/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Pull request merged**